src/enumerable.cr

```diff
@@ -1384,6 +1384,18 @@ module Enumerable(T)
     result
   end
 
+  # Returns a `Hash` with counts the elements in the collection.
+  #
+  # ```
+  # %w[1 2 3 3 3 2].tally # => { "1" => 1, "2" => 2, "3" => 3 }
+  # ```
+  def tally
+    each_with_object(Hash(T, Int32).new) do |item, hash|
+      hash[item] ||= 0
```

Comment:
I would do:

```
count = hash[item]?
if count
  hash[item] = count + 1
else
  hash[item] = 1
end
```

That should be a bit more efficient to avoid multiple lookups in `hash`.

Reply:
Thank you! Updated with your one https://github.com/crystal-lang/crystal/pull/7921/commits/c3e69663b2e579868f01030794051e130c71d945